contract/lib/dns.ts

```diff
@@ -44,11 +33,11 @@ export const debugPrintRecord = (record: object, bufferWriter: Buffer) => {
 
 export const encodeRecord = (record: object): EncodedRecord => {
   const buffer = DNSRecord.write(new BufferWriter(), record).dump() as Buffer
-  return [buffer.toString('hex'), buffer, record]
+  return { hexString: buffer.toString('hex'), buffer, record }
 }
 
 // TODO: buggy for empty ipAddress. Must fix later
-export const encodeARecord = (name: string, ipAddress: string): EncodedRecord => {
```

Comment:
The value is a single type, there is no reason to wrap it in an object. Same as others. I already added whatever types that were necessary

Reply:
Reverted 🙏